x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/accesscontrol/FieldSubsetReader.java

```diff
@@ -371,37 +375,47 @@ private Terms wrapTerms(Terms terms, String field) {
      * representing fields that should not be visible in this reader.
      */
     class FieldNamesTerms extends FilterTerms {
+        long size = 0;
+        long sumDocFreq;
+        int docCount;
 
-        FieldNamesTerms(Terms in) {
+
+        FieldNamesTerms(Terms in) throws IOException {
             super(in);
+            assert in.hasFreqs() == false;
+            // re-compute the stats for the field to take
+            // into account the filtered terms.
+            computeFilteredStats();
         }
 
         @Override
         public TermsEnum iterator() throws IOException {
             return new FieldNamesTermsEnum(in.iterator());
         }
 
-        // we don't support field statistics (since we filter out terms)
-        // but this isn't really a big deal: _field_names is not used for ranking.
-
-        @Override
-        public int getDocCount() throws IOException {
-            return -1;
+        private void computeFilteredStats() throws IOException {
+            TermsEnum e = iterator();
+            while (e.next() != null) {
+                size ++;
+                sumDocFreq += e.docFreq();
+                docCount = Math.max(e.docFreq(), docCount);
```

Comment:
I don't think this is correct... Maybe we should assume docCount = maxDoc.

Reply:
oups thanks, I changed it to return maxDoc instead